interfaces/Portalicious/src/app/components/dynamic-form/dynamic-form.component.html

```diff
@@ -0,0 +1,50 @@
+<form
+  class="block w-full"
+  [formGroup]="formGroup()"
+  (ngSubmit)="onFormSubmit()"
+>
+  <div class="flex flex-col gap-2">
+    @for (field of fields(); track field.controlName) {
+      <!-- eslint-disable-next-line @angular-eslint/template/label-has-associated-control -->
+      <label>
+        <ng-container>{{ field.label }}</ng-container>
+        @switch (field.type) {
+          @case ('password') {
+            <p-password
+              [formControlName]="field.controlName"
+              [autocomplete]="field.autocomplete"
+              [toggleMask]="true"
+              [feedback]="false"
+              styleClass="mt-2 w-full [&_input]:w-full"
+            />
```

Comment:
`<p-password
  [formControlName]="field.controlName"
  [autocomplete]="field.autocomplete"
  [toggleMask]="true"
  [feedback]="false"
  styleClass="mt-2 w-full [&_input]:w-full"
/>`

This could become its own component?
I am saying this because these kind of dynamic components can become monsters.
Also, let's try to have the least amount of exceptions (special fields that need an element other than the default)

Reply:
> This could become its own component?

In this case, I don't see the advantage of doing that.

In general, my preference is to abstract only when reusability is necessary. A good example of a "rule" is the [rule of three](https://en.wikipedia.org/wiki/Abstraction_principle_(computer_programming)#:~:text=A%20rule%20of%20thumb%20governing,needs%20to%20be%20abstracted%20out.). According to this rule, we wouldn't abstract because we are only using this in one place.

But, rules can have exceptions. Let's analyze this particular example. 

I assume, with the extrapolation that you suggest, the code would become something like this:

```html
            <app-dynamic-form-password
              [formControlName]="field.controlName"
              [autocomplete]="field.autocomplete"
              styleClass="mt-2 w-full [&_input]:w-full"
            />
```

We would have a lot more code (two new files, a new class/component definition, together with re-defining input props etc.) but I don't necessarily see the advantage. 

> I am saying this because these kind of dynamic components can become monsters.

I agree that dynamic components can become monsters. But I don't think moving this code elsewhere will improve it (if anything, the opposite). I think there might be other ingredients that will make it a monster with time.

> Also, let's try to have the least amount of exceptions (special fields that need an element other than the default)

I think this is unavoidable, at least with types. I imagine that we will at least have types like radio, select, textarea, date etc. at a certain point.

If we don't want that, then maybe we shouldn't have a dynamic component. That might be the right approach, but then I'm not sure how to avoid the original duplication that this PR is addressing 😬 (and maybe that's ok, and we just need to figure something else out).

Reply:
Poking around the figma files and I see something like this:

![image](https://github.com/user-attachments/assets/7873da8a-bb09-40c5-8b69-490d837a5d73)

And it would require a lot more edge cases and options to work with that layout. I am, once again, questioning everything 😄